### PR TITLE
Add NUM2LL

### PR DESCRIPTION
**Pull request by bjfish**
Used in `pg`.

### CODE REVIEW

File: lib/cext/ruby.h [hunk 1]
```diff
@@ -450,6 +450,15 @@ void rb_num_zerodiv(void);
...
 #define NUM2SSIZET(x) NUM2LONG(x)
 
+#define LLONG_MIN_MINUS_ONE ((double)LLONG_MIN-1)
+#define LLONG_MAX_PLUS_ONE (2*(double)(LLONG_MAX/2+1))
+LONG_LONG rb_num2ll(VALUE val);
+#define NUM2LL(x) rb_num2ll(x)
+#define LLONG_MIN_MINUS_ONE_IS_LESS_THAN(n) \
+  (LLONG_MIN_MINUS_ONE == (double)LLONG_MIN ? \
+   LLONG_MIN <= (n): \
+   LLONG_MIN_MINUS_ONE < (n))
```
Comment (eregon) [inline lib/cext/ruby.h]: These macros, except NUM2LL, should be part of ruby.c.
In MRI, they are defined in numeric.c.
Only public functions/macros from MRI include should be part of the `ruby.h` (and the needed `rb_tr*` debug functions).

Comment (bjfish) [inline lib/cext/ruby.h]: Fixed.

File: src/main/c/cext/ruby.c [hunk 2]
```diff
@@ -418,6 +418,58 @@ unsigned long rb_num2ulong(VALUE val) {
   return (unsigned long)truffle_invoke_l(RUBY_CEXT, "rb_num2ulong", val);
 }
 
+static char * out_of_range_float(char (*pbuf)[24], VALUE val) {
```
Comment (eregon) [inline src/main/c/cext/ruby.c]: The `*` should be next to the function name for consistency (silly C style).

Comment (bjfish) [inline src/main/c/cext/ruby.c]: Fixed.

**Pull request merged**